R/guide-legend.r

```diff
@@ -247,9 +254,16 @@ guide_geom.legend <- function(guide, layers, default_mapping) {
   guide$geoms <- plyr::llply(layers, function(layer) {
     matched <- matched_aes(layer, guide, default_mapping)
 
+    layer$show.legend <- unlist(rename_aes(as.list(layer$show.legend)))
+
     if (length(matched) > 0) {
       # This layer contributes to the legend
-      if (is.na(layer$show.legend) || layer$show.legend) {
+      include <- ifelse(
```

Comment:
I think this should be an `if` statement with separate then and else blocks. Can you please rewrite?